api/rest/client/request.go

```diff
@@ -10,15 +10,42 @@ import (
 	"github.com/ipfs/ipfs-cluster/api"
 )
 
-func (c *Client) do(method, path string, body io.Reader, obj interface{}) error {
-	resp, err := c.doRequest(method, path, body)
+type responseDecoder func(d *json.Decoder) error
```

Comment:
only if we decided to support several output formats to be handled by different decoders. Since our API produces json, I like that it's explicit here. Otherwise I'd like to make this change on the API side too, using multicodecs for encoding, start handling Accept headers etc -> self-contained -> different PR.